blueprints/data-solutions/data-platform-foundations/04-transformation.tf

```
@@ -62,9 +62,10 @@ module "transf-project" {
     "storage.googleapis.com",
     "storage-component.googleapis.com"
   ])
-  service_encryption_key_ids = {
-    dataflow = [try(local.service_encryption_keys.dataflow, null)]
-    storage  = [try(local.service_encryption_keys.storage, null)]
+  service_agent_encryption_key_ids = {
+    dataflow = compact([var.service_encryption_keys.dataflow])
+    compute  = compact([var.service_encryption_keys.dataflow])
+    storage  = compact([var.service_encryption_keys.storage])
```

```suggestion
    storage  = compact([var.service_encryption_keys.dataflow, var.service_encryption_keys.storage])
```
To follow the pattern in `03-orchestration.tf`?

---

Butthen I'd follow the same approach in `05-datawarehouse.tf` too

---

Dataflow only needs 'dataflow' and 'compute', at least according to the [old dependency map](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/project/service-accounts.tf#L25C1-L25C41)